lib/charms/opensearch/v0/helper_cluster.py

```diff
@@ -177,8 +187,11 @@ def nodes(
                         unit_number=int(obj["name"].split("-")[-1]),
                         temperature=obj.get("attributes", {}).get("temp"),
                     )
-                    nodes.append(node)
-
+                    if (
+                        not only_this_juju_app
+                        or "-".join(obj["name"].split("-")[:-1]) == only_this_juju_app
```

Comment:
See previous comment.
As a side note: a `Node` object has an `app_name` attribute, no need to handle it